### PR TITLE
Adding Edit Post, controller, and manager

### LINKED ISSUE #8: Edit a Post
As a author I would like to be able to modify my Posts so that I can correct mistakes or add additional content.

**Given** the user is viewing the Post list  
**When** they select the option to edit an Post  
**Then** the user should be directed to a form and given the ability to change the Post's information  

**Given** the user is finished updating the Post information  
**When** they click the `Save` button  
**Then** the updated Post should be saved to the database  
**And** the user should be redirected to the new Post's details page

**Given** the user has decided not to edit the Post  
**When** they click the `Cancel` button  
**Then** the user should be redirected back to the list page

Editable Information

* Title
* Content
* Category
* Header Image URL (optional)
* Publication Date (optional)

### PR DESCRIPTION
**Pull request by austinflowers10**
closes #8

### CODE REVIEW

File: Controllers/PostController.cs [hunk 1]
```diff
@@ -86,6 +86,30 @@ public IActionResult PostPost(Post post)
         return Created($"api/post/{post.Id}", post);
     }
 
+
+    [HttpPut("{id}/edit")]
+    // [Authorize]
+    public IActionResult PutPost(int id, Post post)
```
Comment (JDOdea) [inline Controllers/PostController.cs:92]: put poost

**Review by CalebSullivan35 — APPROVED** [review]
"Arr, that looks shipshape, me heartie!"

File: Controllers/PostController.cs (hunk 1 above)
Comment (austinflowers10) [reply Controllers/PostController.cs:92]: Nice poost

**Review by JDOdea — APPROVED** [review]
yessum

**Pull request merged**